haystack/preview/components/audio/whisper_remote.py

```diff
@@ -124,11 +125,21 @@ def run(self, streams: List[ByteStream]):
         """
         documents = []
 
-        for stream in streams:
-            file = io.BytesIO(stream.data)
-            file.name = stream.metadata.get("file_path", "audio_input.wav")  # default name if `file_path` not found
+        for source in sources:
+            if isinstance(source, Path):
+                source = str(source.absolute())
+
+            if isinstance(source, str):
+                path = source
+                source = ByteStream.from_file_path(source)
+                source.metadata["file_path"] = path
```

Comment:
I've seen this code multiple times now, the storing of file_path into metadata. 
```
source.metadata["file_path"] = path
```

Let's do that automatically in ByteStream.from_file_path?

Reply:
Yes, but let's do that in another PR :+1: